psi4/driver/procrouting/response/scf_response.py

```diff
@@ -425,6 +425,126 @@ def _validate_tdscf(*, wfn, states, triplets, guess) -> None:
     if guess != "DENOMINATORS":
         raise ValidationError(f"TDSCF: Guess type {guess} is not valid")
 
+# Generate additional output from TDSCF caclulations
+def _analyze_tdscf_excitations(tdscf_results,
+                              wfn,
+                              tda,
+                              coeff_cutoff,
+                              tdm_print):
+
+    restricted = wfn.same_a_b_orbs()
+
+    if 'E_TDM_LEN' in tdm_print:
+    # Print out ETDM vectors (length representation)
+        core.print_out("\nElectric Transition Dipole Moments (Length) (AU):\nState      X          Y          Z\n")
+        for i, q in enumerate(tdscf_results):
+            x, y, z = q.edtm_length
+            core.print_out(f" {i+1: 4d} {x:< 10.6f} {y:< 10.6f} {z:< 10.6f}\n")
+    if 'E_TDM_VEL' in tdm_print:
+    # Print out ETDM vectors (velocity representation)
+        core.print_out("\nElectric Transition Dipole Moments (Velocity) (AU):\nState      X          Y          Z\n")
+        for i, q in enumerate(tdscf_results):
+            x, y, z = q.edtm_velocity
+            core.print_out(f" {i+1: 4d} {x:< 10.6f} {y:< 10.6f} {z:< 10.6f}\n")
+    if 'M_TDM' in tdm_print:
+    # Print out MTDM vectors
+        core.print_out("\nMagentic Transition Dipole Moments (AU):\nState      X          Y          Z\n")
+        for i, q in enumerate(tdscf_results):
+            x, y, z = q.mdtm
+            core.print_out(f" {i+1: 4d} {x:< 10.6f} {y:< 10.6f} {z:< 10.6f}\n")
+
```

Comment:
I suggest a slight refactoring here, such that adding other transition moments later doesn't require too much boilerplate:

```python3
_printable = {
"E_TDM_LEN": {"title": "Electric Transition Dipole Moments (Length) (AU)", "what": lambda x: x.edtm_length, },
"E_TDM_VEL": {"title": "Electric Transition Dipole Moments (Velocity) (AU)",  "what": lambda x: x.edtm_velocity, },
"M_TDM": {"title": "Magnetic Transition Dipole Moments (AU)", "what": lambda x: x.mdtm, },
}

for p in tdm_print:
     core.print_out(f"\n{_printable['p']['title']:\nState      X          Y          Z\n")
     for i, q in enumerate(tdscf_results):
          x, y, z = _printable['p']['what'](q)
          core.print_out(f" {i+1: 4d} {x:< 10.6f} {y:< 10.6f} {z:< 10.6f}\n")
```

Reply:
Oooo...I like that. Done (with a couple of corrections).

I've also run (my new bits of code) through yapf and made the suggested changes (there were a lot suggested for the rest of scf_response.py but that's probably for someone else to sort out).